### ISSUE TITLE
PATCH to fix ilmbase 2.3.0 compatibility in lib/IlmCtlSimd/CtlSimdInst.cpp

### ISSUE THREAD

**Issue by mandree** [ampas/CTL#71]
ilmbase and OpenEXR 2.3.0 change `Iex::BaseExc` to no longer derive from `std::string`, mentioned in the release notes here: [https://github.com/openexr/openexr/releases/tag/v2.3.0](https://github.com/openexr/openexr/releases/tag/v2.3.0)

This causes a compilation error on `lib/IlmCtlSimd/CtlSimdInst.cpp` when it propagates a caught exception, because there is no longer an implicit conversion from `Iex::BaseExc` to `std::string` available. 

It *appears* as though the fix were to use `e.what()` instead of `e`, which is what the attached patch does.
[patch-lib_IlmCtlSimd_CtlSimdInst.cpp.txt](https://github.com/ampas/CTL/files/2402792/patch-lib_IlmCtlSimd_CtlSimdInst.cpp.txt)

**Comment by mandree** [comment]
This is the inline patch for perusal: 

```
--- lib/IlmCtlSimd/CtlSimdInst.cpp.orig	2014-06-03 01:11:24 UTC
+++ lib/IlmCtlSimd/CtlSimdInst.cpp
@@ -189,7 +189,7 @@ SimdInst::executePath (SimdBoolMask &mas
 	    REPLACE_EXC
 		(e, "\n" <<
 		 xcontext.fileName() << ":" <<
-		 inst->lineNumber() << ": " << e);
+		 inst->lineNumber() << ": " << e.what());
 
 	    throw e;
 	}
```

**Comment by remia** [comment]
Thanks for the patch, I had the same issue when trying to update IlmBase / OpenEXR to 2.3.0 (needed to compile with C++17 enabled).

Is it planned to update CTL ?

**Comment by mandree** [comment]
@remia, given that the latest commit has been made more than four years ago, personally I don't currently believe in CTL upgrades, and I am neither member of the development team nor speaking on their behalf, just guessing from observation. I posted the patch where it is more likely to be found.

**Comment by michaeldsmith** [comment]
fixed by [80d175b](https://github.com/ampas/CTL/commit/80d175b5f6888e84aedaca70975b55bcb69a2910)
